blmath/geometry/primitives/plane.py

```diff
@@ -210,25 +210,22 @@ def polyline_xsection(self, polyline):
         return intersection_points
 
     def line_xsection(self, pt, ray):
-        pt = np.asarray(pt).ravel()
-        ray = np.asarray(ray).ravel()
-        assert len(pt) == 3
-        assert len(ray) == 3
+        return self._line_xsection(np.asarray(pt).ravel(), np.asarray(ray).ravel())
+
+    def _line_xsection(self, pt, ray):
         denom = np.dot(ray, self.normal)
         if denom == 0:
             return None # parallel, either coplanar or non-intersecting
         p = np.dot(self.reference_point - pt, self.normal) / denom
         return p * ray + pt
 
     def line_segment_xsection(self, a, b):
-        a = np.asarray(a).ravel()
-        b = np.asarray(b).ravel()
-        assert len(a) == 3
-        assert len(b) == 3
+        return self._line_segment_xsection(np.asarray(a).ravel(), np.asarray(b).ravel())
 
-        pt = self.line_xsection(a, b-a)
+    def _line_segment_xsection(self, a, b):
+        pt = self._line_xsection(a, b-a)
         if pt is not None:
-            if np.any(pt < np.min((a, b), axis=0)) or np.any(pt > np.max((a, b), axis=0)):
+            if any(np.logical_and(pt > a, pt > b)) or any(np.logical_and(pt < a, pt < b)):
```

Comment:
Here's an interesting example of a hotspot that profiling detected. We call this function half a million times, so tiny optimizations help a lot. Since numpy is optimized for large arrays, something like `np.min` is not as efficient as you'd expect when called many many times on tiny arrays.